### PR TITLE
dev-59: Implemented basic water shader

### PR DESCRIPTION
**Pull request by StijnKlopper**
Implemented basic water shader for ocena based on the unity standard assets:
https://assetstore.unity.com/packages/essentials/asset-packs/standard-assets-for-unity-2018-4-32351

### CODE REVIEW

**Comment by StijnKlopper** [comment]
Video of shader in action
https://user-images.githubusercontent.com/33661926/104369914-6c57b800-551e-11eb-92f0-d6b447dd5aac.mp4

File: No FYS-X Sky/Assets/Helpers/WaterScripts/SpecularLighting.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        public Transform specularLight;
+        private WaterBase m_WaterBase;
+
+
+        public void Start()
+        {
+            m_WaterBase = (WaterBase)gameObject.GetComponent(typeof(WaterBase));
+        }
+
```
Comment (appelmoesje) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/SpecularLighting.cs:18]: Remove whitespace

File: No FYS-X Sky/Assets/Helpers/WaterScripts/MeshContainer.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+using System;
+using UnityEngine;
+
+namespace UnityStandardAssets.Water
```
Comment (Fakerick1) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/MeshContainer.cs:4]: We dont use namespaces in the rest of the project

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    Camera.main.depthTextureMode |= DepthTextureMode.Depth;
+                }
+            }
+            else
+            {
+                Shader.EnableKeyword("WATER_EDGEBLEND_OFF");
+                Shader.DisableKeyword("WATER_EDGEBLEND_ON");
+            }
+        }
+
```
Comment (appelmoesje) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs:59]: Remove white whitespace

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public class WaterTile : MonoBehaviour
+    {
+        public WaterBase waterBase;
+
+
+        public void Start()
+        {
+            AcquireComponents();
+        }
+
```
Comment (appelmoesje) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs:16]: Remove white whitespace

File: No FYS-X Sky/Assets/Helpers/WaterScripts/MeshContainer.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public class MeshContainer
+    {
+        public Mesh mesh;
```
Comment (Fakerick1) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/MeshContainer.cs:8]: public starts with a capital letter

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                {
+                    waterBase = transform.parent.GetComponent<WaterBase>();
+                }
+                else
+                {
+                    waterBase = transform.GetComponent<WaterBase>();
+                }
+            }
+        }
+
```
Comment (appelmoesje) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs:33]: Remove white whitespace

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        public void WaterTileBeingRendered(Transform tr, Camera currentCam)
+        {
+            if (currentCam && edgeBlend)
+            {
+                currentCam.depthTextureMode |= DepthTextureMode.Depth;
+            }
+        }
+
+
```
Comment (appelmoesje) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs:69]: Remove white whitespace

**Review by appelmoesje — CHANGES_REQUESTED** [review]

File: No FYS-X Sky/Assets/Helpers/WaterScripts/SpecularLighting.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
+using System;
+using UnityEngine;
+
+namespace UnityStandardAssets.Water
+{
+    [RequireComponent(typeof(WaterBase))]
+    [ExecuteInEditMode]
+    public class SpecularLighting : MonoBehaviour
+    {
+        public Transform specularLight;
```
Comment (Fakerick1) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/SpecularLighting.cs:10]: public starts with capital

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        High = 2,
+        Medium = 1,
+        Low = 0,
+    }
+
+    [ExecuteInEditMode]
+    public class WaterBase : MonoBehaviour
+    {
+        public Material sharedMaterial;
```
Comment (Fakerick1) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs:16]: public starts with capital

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        public WaterQuality waterQuality = WaterQuality.High;
+        public bool edgeBlend = true;
```
Comment (Fakerick1) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterBase.cs:18]: image

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+using UnityEngine;
+
+namespace UnityStandardAssets.Water
+{
+    [ExecuteInEditMode]
+    public class WaterTile : MonoBehaviour
+    {
+        public WaterBase waterBase;
+
+
```
Comment (Fakerick1) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs:11]: Remove white whitespace

File: No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+        public void Start()
+        {
+            AcquireComponents();
+        }
+
+
+        void AcquireComponents()
+        {
+
```
Comment (Fakerick1) [inline No FYS-X Sky/Assets/Helpers/WaterScripts/WaterTile.cs:20]: Remove white whitespace

**Review by Fakerick1 — CHANGES_REQUESTED** [review]
Looks good bro

**Review by Fakerick1 — APPROVED** [review]

**Pull request merged**